src/modm/platform/usb/sam/usb.hpp.in

```diff
@@ -0,0 +1,48 @@
+/*
+* Copyright (c) 2020, Erik Henriksson
+* Copyright (c) 2020, Niklas Hauser
+*
+* This file is part of the modm project.
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <modm/platform/device.hpp>
+#include <modm/platform/gpio/connector.hpp>
+
+namespace modm::platform
+{
+
+class Usb
+{
+public:
+	template< class SystemClock >
+	static void
+	initialize()
+	{
+		static_assert(SystemClock::Usb == 48_MHz, "Usb must have a 48MHz clock!");
```

Comment:
This should be `SystemClock::Frequency` because we are using the system clock generator below. The current code has an implicit assumption that `SystemClock::Usb == SystemClock::Frequency`, but that is not statically checked anywhere.

Note: SAM supports clocking the USB peripheral from an arbitrary clock generator, so we could get rid of this restriction by passing the clock generator as a template arg. That said, probably no need as nobody is running SAM devices at a frequency != 48MHz right now...

Reply:
Yeah, this is the knowledge that I lack. Good thing I can ping you ;-P